File: android/app/src/main/java/com/liive/ride/ui/RideSheetLayout.kt
package com.liive.ride.ui

import com.liive.ride.designsystem.LiiveControl
import com.liive.ride.designsystem.LiiveSpacing
import com.liive.ride.designsystem.LiiveStroke

internal object RideSheetLayout {
    val hairlineHeight = LiiveStroke.hairline
    val selectedBorderWidth = LiiveStroke.active

    val compactGap = LiiveSpacing.xs2
    val inlineGap = LiiveSpacing.xs + LiiveSpacing.xs2
    val controlGap = LiiveSpacing.s
    val rowGap = LiiveSpacing.s + LiiveSpacing.xs2
    val headerBottomPadding = LiiveSpacing.m
    val sectionGap = LiiveSpacing.m + LiiveSpacing.xs2

    val searchHeight = LiiveControl.md + LiiveSpacing.xs2
    val searchHorizontalPadding = sectionGap
    val searchIconSize = LiiveSpacing.xl - LiiveSpacing.xs2

    val backButtonSize = LiiveControl.sm
    val backIconSize = LiiveSpacing.xl
    val savedPlaceIconSize = LiiveControl.md - LiiveSpacing.s
    val optionIconSize = LiiveControl.sm
    val paymentStatusIconSize = LiiveControl.md - LiiveSpacing.s
    val receiptIconSize = LiiveControl.xl

    val tierSpacing = LiiveSpacing.s
    val tierRowPadding = LiiveSpacing.m

    val matchingDotCount = 3
    val matchingDotSize = LiiveSpacing.s + LiiveSpacing.xs2 / 2
    val matchingDotLift = -(LiiveSpacing.s - LiiveSpacing.xs2 / 2)
    const val matchingDotDurationMs = 600
    const val matchingDotDelayMs = 160
    val matchingDescriptionMaxWidth = LiiveControl.xl * MaxWidthColumnCount
    val matchingContentTopPadding = LiiveSpacing.s
    val matchingContentBottomPadding = LiiveSpacing.xxl - LiiveSpacing.xs2

    val multiLegPanelPadding = sectionGap
    val multiLegPanelTopPadding = LiiveSpacing.m
    val multiLegIconSize = LiiveSpacing.l
    val transferIconSize = LiiveSpacing.l - LiiveSpacing.xs2 / 2
    val progressSeparatorTrackHeight = LiiveSpacing.xs2 + LiiveSpacing.xs2 / 4

    val receiptContentVerticalPadding = rowGap
    val receiptButtonTopPadding = LiiveSpacing.xl
    val fareCardHorizontalPadding = sectionGap
    val fareCardTopPadding = LiiveSpacing.s
    val fareCardBottomPadding = sectionGap
    val paymentSectionGap = LiiveSpacing.m
    val securityCopyTopPadding = rowGap

    val ratingTopPadding = LiiveSpacing.m
    val ratingBottomPadding = LiiveSpacing.l
    val ratingStarSize = LiiveSpacing.xxxl - LiiveSpacing.xs
    val ratingStarPadding = LiiveSpacing.xs2

    val sosMessageTopPadding = rowGap
    val sosMessageBottomPadding = LiiveSpacing.xl - LiiveSpacing.xs2
    val sosButtonGap = LiiveSpacing.s
    val sosPanelPadding = LiiveSpacing.xxl - LiiveSpacing.xs2
    val sosPanelMaxWidth = matchingDescriptionMaxWidth + LiiveSpacing.xl

    private const val MaxWidthColumnCount = 5f
}

File: Packages/RideSharingFeature/Sources/RideSharingFeature/RideSheetLayout.swift
import SwiftUI

enum RideSheetLayout {
    static let stackedSpacing: CGFloat = 0
    static let hairlineHeight: CGFloat = 0.5
    static let selectedBorderWidth: CGFloat = 1.5

    static let compactGap = LiiveSpacing.xs2
    static let inlineGap = LiiveSpacing.xs + LiiveSpacing.xs2
    static let controlGap = LiiveSpacing.s
    static let rowGap = LiiveSpacing.s + LiiveSpacing.xs2
    static let headerBottomPadding = LiiveSpacing.m
    static let sectionGap = LiiveSpacing.m + LiiveSpacing.xs2

    static let searchHeight = LiiveControl.md + LiiveSpacing.xs2
    static let searchHorizontalPadding = sectionGap
    static let searchIconSize = LiiveSpacing.xl - LiiveSpacing.xs2

    static let backButtonSize = LiiveControl.sm
    static let backIconSize = LiiveSpacing.xl
    static let savedPlaceIconSize = LiiveControl.md - LiiveSpacing.s
    static let optionIconSize = LiiveControl.sm
    static let paymentStatusIconSize = LiiveControl.md - LiiveSpacing.s
    static let receiptIconSize = LiiveControl.xl

    static let tierSpacing = LiiveSpacing.s
    static let tierRowPadding = LiiveSpacing.m

    static let matchingDotCount = 3
    static let matchingDotSize = LiiveSpacing.s + LiiveSpacing.xs2 / 2
    static let matchingDotLift = -(LiiveSpacing.s - LiiveSpacing.xs2 / 2)
    static let matchingDotDuration: Double = 0.6
    static let matchingDotDelay: Double = 0.16
    static let matchingDescriptionMaxWidth = LiiveControl.xl * maxWidthColumnCount
    static let matchingContentTopPadding = LiiveSpacing.s
    static let matchingContentBottomPadding = LiiveSpacing.xxl - LiiveSpacing.xs2

    static let multiLegPanelPadding = sectionGap
    static let multiLegPanelTopPadding = LiiveSpacing.m
    static let multiLegIconSize = LiiveSpacing.l
    static let transferIconSize = LiiveSpacing.l - LiiveSpacing.xs2 / 2
    static let transferSeparatorTopPadding = LiiveSpacing.xs2

    static let receiptContentVerticalPadding = rowGap
    static let receiptButtonTopPadding = LiiveSpacing.xl
    static let fareCardHorizontalPadding = sectionGap
    static let fareCardTopPadding = LiiveSpacing.s
    static let fareCardBottomPadding = sectionGap
    static let paymentSectionGap = LiiveSpacing.m
    static let securityCopyTopPadding = rowGap

    static let ratingBottomPadding = LiiveSpacing.l
    static let ratingStarSize = LiiveSpacing.xxxl - LiiveSpacing.xs
    static let ratingStarPadding = LiiveSpacing.xs2

    static let sosMessageTopPadding = rowGap
    static let sosMessageBottomPadding = LiiveSpacing.xl - LiiveSpacing.xs2
    static let sosButtonGap = LiiveSpacing.s
    static let sosPanelPadding = LiiveSpacing.xxl - LiiveSpacing.xs2
    static let sosPanelMaxWidth = matchingDescriptionMaxWidth + LiiveSpacing.xl

    private static let maxWidthColumnCount: CGFloat = 5
}
